projects/packages/my-jetpack/_inc/hooks/use-notification-watcher/use-notification-watcher.ts

```diff
@@ -0,0 +1,123 @@
+import { __, sprintf } from '@wordpress/i18n';
+import { useContext, useEffect } from 'react';
+import { MyJetpackRoutes } from '../../constants';
+import { NOTICE_PRIORITY_MEDIUM, NOTICE_PRIORITY_HIGH } from '../../context/constants';
+import { NoticeContext } from '../../context/notices/noticeContext';
+import { useAllProducts } from '../../data/products/use-product';
+import { getMyJetpackWindowInitialState } from '../../data/utils/get-my-jetpack-window-state';
+import getProductSlugsThatRequireUserConnection from '../../data/utils/get-product-slugs-that-require-user-connection';
+import useAnalytics from '../use-analytics';
+import useMyJetpackNavigate from '../use-my-jetpack-navigate';
+
+type RedBubbleAlerts = Window[ 'myJetpackInitialState' ][ 'redBubbleAlerts' ];
+
+const useNotificationWatcher = () => {
+	const { redBubbleAlerts } = getMyJetpackWindowInitialState();
+
+	useBadInstallNotice( redBubbleAlerts );
+	useSiteConnectionNotice( redBubbleAlerts );
+};
+
+export default useNotificationWatcher;
+
+const useSiteConnectionNotice = ( redBubbleAlerts: RedBubbleAlerts ) => {
+	const { setNotice } = useContext( NoticeContext );
+	const products = useAllProducts();
+	const navToConnection = useMyJetpackNavigate( MyJetpackRoutes.Connection );
+
+	useEffect( () => {
+		if ( ! Object.keys( redBubbleAlerts ).includes( 'missing-site-connection' ) ) {
+			return;
+		}
+
+		const productSlugsThatRequireUserConnection =
+			getProductSlugsThatRequireUserConnection( products );
+
+		const oneProductMessage = sprintf(
+			/* translators: placeholder is product name. */
+			__(
+				'Jetpack %s needs a user connection to WordPress.com to be able to work.',
+				'jetpack-my-jetpack'
+			),
+			productSlugsThatRequireUserConnection[ 0 ]
+		);
+
+		const needsUserConnectionMessage =
+			productSlugsThatRequireUserConnection.length === 1
+				? oneProductMessage
+				: __(
+						'Some products need a user connection to WordPress.com to be able to work.',
+						'jetpack-my-jetpack'
+				  );
+
+		const noticeOptions = {
+			status: 'warning',
+			actions: [
+				{
+					label: __( 'Connect your user account to fix this', 'jetpack-my-jetpack' ),
+					onClick: navToConnection,
+					noDefaultClasses: true,
+				},
+			],
+		};
+
+		setNotice( {
+			message: needsUserConnectionMessage,
+			options: noticeOptions,
+			priority: NOTICE_PRIORITY_HIGH,
+		} );
+	}, [ navToConnection, products, redBubbleAlerts, setNotice ] );
+};
+
+const useBadInstallNotice = ( redBubbleAlerts: RedBubbleAlerts ) => {
+	const { setNotice } = useContext( NoticeContext );
+	const { recordEvent } = useAnalytics();
+
+	useEffect( () => {
+		const badInstallAlerts = Object.keys( redBubbleAlerts ).filter( key =>
+			key.endsWith( '-bad-installation' )
+		) as Array< ` ${ string }-bad-installation` >;
```

Comment:
I believe the leading space is unintentional, right?

```suggestion
		) as Array< `${ string }-bad-installation` >;
```

Reply:
Yes it is, thank you